src/librustc_parse/parser/path.rs

```diff
@@ -71,7 +71,23 @@ impl<'a> Parser<'a> {
             debug!("parse_qpath: (decrement) count={:?}", self.unmatched_angle_bracket_count);
         }
 
-        self.expect(&token::ModSep)?;
+        let lo_colon = self.token.span;
+        if self.eat(&token::Colon) {
```

Comment:
```suggestion
        if self.token.kind == token::Colon {
```
Using `.eat(...)` will add an expectation which produces sub-optimal diagnostics.
After changing this you'll need `self.bump();` instead.

Reply:
The failing test is failing due to this:

```
2020-01-30T23:39:56.6394695Z - error: expected `::`, found `(`
2020-01-30T23:39:56.6394804Z + error: expected one of `::` or `:`, found `(`
2020-01-30T23:39:56.6395173Z 3    |
2020-01-30T23:39:56.6395173Z 3    |
2020-01-30T23:39:56.6395271Z 4 LL |     vec![Ok(2)].into_iter().collect:<Result<Vec<_>,_>>()?;
2020-01-30T23:39:56.6395638Z -    |                                    -                  ^ expected `::`
2020-01-30T23:39:56.6395931Z +    |                                    -                  ^ expected one of `::` or `:`
2020-01-30T23:39:56.6396061Z 6    |                                    |
2020-01-30T23:39:56.6396164Z 7    |                                    help: maybe write a path separator here: `::`
2020-01-30T23:39:56.6396164Z 7    |                                    help: maybe write a path separator here: `::`
2020-01-30T23:39:56.6396283Z 8    |
2020-01-30T23:39:56.6396363Z 
```